0000-template.md

```diff
@@ -0,0 +1,110 @@
+Explain the proposal as if it was already included in the language and you were teaching it to another Amethyst programmer. That generally means:
+
+- Introducing new named concepts.
+- Explaining the feature largely in terms of examples.
+- Explaining how Amethyst developers should *think* about the feature, and how it should impact the way they use Amethyst. It should explain the impact as concretely as possible.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing Amethyst programmers and new Amethyst programmers.
+
+For implementation-oriented RFCs (e.g. for changes to the engine), this section should focus on how engine contributors should think about the change, and give examples of its concrete impact. For policy RFCs, this section should provide an example-driven introduction to the policy, and explain its impact in concrete terms.
+</details>
+
+# Reference-Level Explanation
+[reference-level-explanation]: #reference-level-explanation
+<details>
+<summary>The technical details and design of the RFC.</summary>
+This is the technical portion of the RFC. Explain the design in sufficient detail that:
+
+- Its interaction with other features is clear.
+- It is reasonably clear how the feature would be implemented.
+- Corner cases are dissected by example.
+
+The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work.
+</details>
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+# Rationale and Alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not choosing them?
+- What is the impact of not doing this?
+
+# Prior Art
+[prior-art]: #prior-art
+<details>
+<summary>Discuss previous attempts, both good and bad, and how they relate to this proposal.</summary>
+A few examples of what this can include are:
+
+- For engine, network, web, and rendering proposals: Does this feature exist in other engines and what experience has their community had?
+- For community proposals: Is this done by some other community and what were their experiences with it?
+- For other teams: What lessons can we learn from what other communities have done here?
+- Papers: Are there any published papers or great posts that discuss this? If you have some relevant papers to refer to, this can serve as a more detailed theoretical background.
+
+This section is intended to encourage you as an author to think about the lessons from other languages, provide readers of your RFC with a fuller picture.
+If there is no prior art, that is fine - your ideas are interesting to us whether they are brand new or if it is an adaptation from other languages.
```

Comment:
Should languages here be engines?

Reply:
@Aceeri Hmm, not sure what you mean. An engine might implement something from a paper (a new consensus algorithm, gossip protocol, etc). Or do you mean they might reference an engine where the feature already exists? If so, I can add that in.

Reply:
Oh, I misread. You are right.

Reply:
Fixed